File: Graph(그래프)/BFS(너비 우선 탐색)/백준_1697_숨바꼭질/숨바꼭질.cs
using System.Collections;
using static System.Console;
using System;

class Program
{
    static int a, b;
    static Queue q = new Queue();
    static ArrayList _move = new ArrayList();
    static bool[] check = new bool[100001];
    static int[] seconds = new int[100001];
    static void BFS(int n)
    {
        q.Enqueue(n);
        check[n] = true;
        while (q.Count != 0)
        {
            int f = (int)q.Dequeue();
            _move.Add(f - 1);
            if (f < b)
            {
                _move.Add(f + 1);
                _move.Add(f * 2);
            }
            foreach (int i in _move)
            {
                if (i >= 0 && i <= 100000)
                {
                    if (check[i] == false)
                    {
                        q.Enqueue(i);
                        check[i] = true;
                        seconds[i] = seconds[f] + 1;
                        if (i == b) return;
                    }
                }
            }
            _move.Clear();
        }
    }

    public static void Main()
    {
        string[] s = ReadLine().Split(' ');
        a = int.Parse(s[0]);
        b = int.Parse(s[1]);

        BFS(a);
        Write(seconds[b]);

    }
}

File: Graph(그래프)/BFS(너비 우선 탐색)/백준_1697_숨바꼭질/숨바꼭질.py
from collections import deque

q = deque()
a, b = map(int, input().split(' '))
_move = []
check = [False for _ in range(200001)]      #파이썬만 100001할 시, indexerror
seconds = [0 for _ in range(200001)]

def bfs(n):
    q.append(n)
    check[n] = True
    while len(q) != 0:
        f = q.popleft()
        _move.append(f-1)
        if f < b:
            _move.append(f+1)
            _move.append(f*2)
        for i in _move:
            if i >= 0 and i <= 100000:
                if check[i] == False:
                    check[i] = True
                    q.append(i)
                    seconds[i] = seconds[f] + 1
                    if i == b:
                        return
        _move.clear()

bfs(a)
print(seconds[b])